tools/wptrunner/wptrunner/executors/executormarionette.py

```diff
@@ -1269,5 +1269,9 @@ def _render(self, protocol, url, timeout):
 class MarionetteWdspecExecutor(WdspecExecutor):
     def __init__(self, logger, browser, *args, **kwargs):
         super().__init__(logger, browser, *args, **kwargs)
+
+        args = self.capabilities["moz:firefoxOptions"].setdefault("args", [])
+        args.extend(["--profile", self.browser.profile])
```

Comment:
Why aren't we going to use [`moz:firefoxOptions.profile`](https://developer.mozilla.org/en-US/docs/Web/WebDriver/Capabilities/firefoxOptions#profile) and pass a base64 encoded profile? Doesn't this change break our support for Android?

Reply:
Geckodriver does the right thing on android, I think? https://searchfox.org/mozilla-central/source/testing/geckodriver/src/capabilities.rs#422 sets the `profile` field of `FirefoxOptions` if `--profile` is passed. https://searchfox.org/mozilla-central/source/testing/geckodriver/src/browser.rs#184 passes that path down to `handler.prepare` and https://searchfox.org/mozilla-central/source/testing/geckodriver/src/android.rs#366 pushes the profile from the local system to the device.

Reply:
Discussed last week with @jgraham and that is indeed fine. We don't need the base64 encoded profile feature for wpt and as such it's possible to pass profile through the Firefox command line argument, which are then picked-up by geckodriver.